src/main/java/org/folio/service/organization/OrganizationStorageService.java

```diff
@@ -77,112 +79,122 @@ private boolean isSameAccountNumbers(Organization organization) {
         ACCOUNT_NUMBER_MUST_BE_UNIQUE.toError()));
-      return future;
     }
-    HttpClientInterface client = getHttpClient(headers);
-    handleGetRequest(resourceByIdPath(ORGANIZATIONS, id), client, headers, logger)
-      .thenApply(existingOrganizationJson -> existingOrganizationJson.mapTo(Organization.class))
-      .thenCompose(existingOrganization -> protectionService.validateAcqUnitsOnUpdate(updatedOrganization, existingOrganization, lang, context, headers))
-      .thenAccept(ok -> handlePutRequest(resourceByIdPath(ORGANIZATIONS, updatedOrganization.getId()), JsonObject.mapFrom(updatedOrganization), client, headers,
-        logger).handle((org, t) -> {
-        client.closeClient();
+    handleGetRequest(resourceByIdPath(ORGANIZATIONS, id), requestContext, logger)
+      .compose(existingOrganizationJson -> {
+        Organization organization = existingOrganizationJson.mapTo(Organization.class);
+        organizationPromise.complete(organization);
+        return organizationPromise.future();
+      })
+      .compose(existingOrganization -> protectionService.validateAcqUnitsOnUpdate(updatedOrganization, existingOrganization, lang, context, headers)
+      .compose(ok -> handlePutRequest(resourceByIdPath(ORGANIZATIONS, updatedOrganization.getId()), JsonObject.mapFrom(updatedOrganization), logger, requestContext))
+        .compose(ok -> {
+          promise.complete();
+        return promise.future();
+        })
+        .onFailure(t -> {
         if (Objects.nonNull(t)) {
           logger.warn("Error updating organization with id: {}", id, t);
-          future.completeExceptionally(t);
-        } else {
-          future.complete(null);
+          promise.fail(new CompletionException(t));
         }
-        return null;
-      }))
-    .exceptionally(t -> {
-      logger.error("Error updating organization with id: {}", id, t);
-      future.completeExceptionally(t.getCause());
-      return null;
-    });
-    return future;
+        })).onFailure(t -> {
+        if (Objects.nonNull(t)) {
+          logger.warn("Error updating organization with id: {}", id, t);
+          promise.fail(new CompletionException(t));
+        }
+      });
+    return promise.future();
   }
 
   @Override
-  public CompletableFuture<Void> deleteOrganizationById(String id, Context context, Map<String, String> headers) {
+  public Future<Void> deleteOrganizationById(String id, Context context, Map<String, String> headers) {
     logger.debug("deleteOrganizationById:: Trying to delete organization by id: {}", id);
-    HttpClientInterface client = getHttpClient(headers);
-    return handleDeleteRequest(resourceByIdPath(ORGANIZATIONS, id), client, headers, logger).handle((org, t) -> {
-      client.closeClient();
+    RequestContext requestContext = new RequestContext(context, headers);
+    Promise<Void> promise = Promise.promise();
+    return handleDeleteRequest(resourceByIdPath(ORGANIZATIONS, id), requestContext, logger)
+      .compose(t -> {
```

Comment:
This piece of code logically does nothing and should be removed

Reply:
fixed